src/__tests__/geminiService.test.ts

```
@@ -0,0 +1,604 @@
+        if (mockWorker.onmessage) {
+          // Send progress message
+          mockWorker.onmessage(new MessageEvent('message', {
+            data: {
+              type: 'progress',
+              payload: mockProgress
+            }
+          }));
+          
+          // Send success message
+          setTimeout(() => {
+            mockWorker.onmessage!(new MessageEvent('message', {
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)